src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java

```diff
@@ -289,8 +298,14 @@ private int addProbabilisticSplits(
 	private List<InputSplit> filterByInterval(List<InputSplit> splits, Configuration conf)
 			throws IOException {
 		List<Interval> intervals = getIntervals(conf);
+		int maxIntervalsForSplitFiltering = conf.getInt(MAX_INTERVALS_FOR_SPLIT_FILTERING_PROPERTY,
+				DEFAULT_MAX_INTERVALS_FOR_SPLIT_FILTERING);
 		if (intervals == null) {
 			return splits;
+		} else if (intervals.size() > maxIntervalsForSplitFiltering) {
+			System.err.println("INFO: not filtering splits by intervals since number of " +
```

Comment:
Does hadoop bam have a logging framework?  Better to emit a logging message that could be configured by downstream projects than just the system.out dump.